src/elastio-snap.c

```diff
@@ -2705,6 +2734,14 @@ static int bio_make_read_clone(struct bio_set *bs, struct tracing_params *tp, st
 	bio_sector(new_bio) = sect;
 	bio_idx(new_bio) = 0;
 
+#ifdef USE_BDOPS_SUBMIT_BIO
+	/**
+	 * redirect bio to the physical partition to make sure
+	 * the 'dev->sd_sect_off' math remains the same
+	 * */
+	new_bio->bi_bdev = new_bio->bi_bdev->bd_disk->part0;
+#endif
```

Comment:
`USE_BDOPS_SUBMIT_BIO` is defined as a directive to use `block_device_operations` structure and `submit_bio` function it it starting from the kernel version 5.9. It seems to be not the best identifier for the bio redirection to the physical partition.... Also there are these test failures https://github.com/elastio/elastio-snap/runs/6381324074?check_suite_focus=true#step:10:125 on 5.10 because of the different name or location of `bi_bdev`.
Please come up with the new "compat" which will check necessity to use this approach according to the difference between 5.15 and 5.16 kernels (probably). Also, as far as I see, [`struct block_device part0`](https://elixir.bootlin.com/linux/v5.13.18/C/ident/part0) appeared in the Linux kernel starting from the version 5.11. Does it mean, that in some cases this problem with the superblock corruption may appear on the kernels 5.11 - 5.15? Or, in other words, do we need this fix for the kernels 5.11 - 5.15? If yes, then the new compat may check existence of the `struct block_device part0` in the `struct gendisk`.

Another small suggestion is to add kernel version to the comment.